### PR TITLE
Remove broken getvolumename and pass PV or volume name to attach call

### PR DESCRIPTION
**Pull request by chakri-nelluri**
What this PR does / why we need it:
Flex getvolumename is broken in 1.6. It needs to be fixed comprehensively in 1.7 release. Removing the api in 1.6. Also pass PV or volume name to the driver during attach call. Detach uses PV or volume name, so plugin can use that information to map to PV.
Which issue this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged): fixes #
Fixes - #44737

Original PR: #46136
Moving to this PR and rebase to release-1.6

### CODE REVIEW

**Comment by enisoc** [comment]
Approving for cherrypick. Please get regular approval from owners since this is not copied from a merged PR on master.

@k8s-bot ok to test

File: pkg/volume/flexvolume/driver-call.go [hunk 1]
```diff
@@ -168,6 +169,8 @@ func NewOptionsForDriver(spec *volume.Spec, host volume.VolumeHost, extraOptions
 		options[optionReadWrite] = "rw"
 	}
 
+	options[optionPVorVolumeName] = spec.Name()
```
Comment (saad-ali) [inline pkg/volume/flexvolume/driver-call.go]: Why is this necessary?

Comment (chakri-nelluri) [inline pkg/volume/flexvolume/driver-call.go]: Detach API uses "PVOrVolumeName". If we do not pass it during attach API, driver has no way of mapping it to back during Detach Call. Current workaround/limitation is to use the same name for actual driver backed volume name & PVName.

**Comment by saad-ali** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by saad-ali** [comment]
@k8s-bot pull-kubernetes-verify test this

**Comment by saad-ali** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by saad-ali** [comment]
@k8s-bot pull-kubernetes-verify test this

**Comment by saad-ali** [comment]
@k8s-bot pull-kubernetes-verify test this

**Comment by chakri-nelluri** [comment]
@saad-ali @msau42 Please cherry-pick this for master & 1.7 releases too. This is the only change we have on top of 1.6 flex volume changes.
